pkg/master/controller.go

```diff
@@ -117,6 +117,9 @@ func (c *Controller) UpdateKubernetesService() error {
 		if err := c.CreateMasterServiceIfNeeded("kubernetes", c.ServiceIP, servicePorts, serviceType); err != nil {
 			return err
 		}
+		if err := c.SetService("kubernetes", servicePorts, serviceType); err != nil {
```

Comment:
All masters must use the same ports and admins injecting extra ports would need to be aware of this. I've self-doc'd that above the `SetService` function in case someone runs into it - similarly, extra endpoints must match on all api servers as well.